pkg/controller/pxcrestore/controller.go

```diff
@@ -105,370 +103,279 @@ func (r *ReconcilePerconaXtraDBClusterRestore) Reconcile(ctx context.Context, re
 		if err != nil {
-			return rr, errors.Wrap(err, "set status")
+			if errors.Is(err, errWaitValidate) {
+				return rr, nil
+			}
+			err = errors.Wrap(err, "failed to validate restore job")
+			return rr, err
 		}
-
-		err = r.pitr(ctx, cr, bcp, cluster)
+		cr.Status.PXCSize = cluster.Spec.PXC.Size
+		cr.Status.AllowUnsafeConfig = cluster.Spec.AllowUnsafeConfig
+		log.Info("stopping cluster", "cluster", cr.Spec.PXCCluster)
+		statusState = api.RestoreStopCluster
+	case api.RestoreStopCluster:
+		err = stopCluster(ctx, r.client, cluster.DeepCopy())
 		if err != nil {
-			return rr, errors.Wrap(err, "run pitr")
+			switch err {
+			case errWaitingPods, errWaitingPVC:
+				log.Info("waiting for cluster to stop", "cluster", cr.Spec.PXCCluster, "msg", err.Error())
+				return rr, nil
+			}
+			return rr, errors.Wrapf(err, "stop cluster %s", cluster.Name)
 		}
 
-		cluster.Spec.PXC.Size = oldSize
-		cluster.Spec.AllowUnsafeConfig = oldUnsafe
+		log.Info("starting restore", "cluster", cr.Spec.PXCCluster, "backup", cr.Spec.BackupName)
+		if err := createRestoreJob(ctx, r.client, restorer, false); err != nil {
+			if errors.Is(err, errWaitInit) {
+				return rr, nil
+			}
+			err = errors.Wrap(err, "run restore")
+			return rr, err
+		}
+		statusState = api.RestoreRestore
+	case api.RestoreRestore:
+		restorerJob, err := restorer.Job()
+		if err != nil {
+			return rr, errors.Wrap(err, "failed to create restore job")
+		}
+		job := new(batchv1.Job)
+		if err := r.client.Get(ctx, types.NamespacedName{
+			Name:      restorerJob.Name,
+			Namespace: restorerJob.Namespace,
+		}, job); err != nil {
+			return rr, errors.Wrap(err, "failed to get restore job")
+		}
 
-		log.Info("starting cluster", "cluster", cr.Spec.PXCCluster)
-		err = r.setStatus(cr, api.RestoreStartCluster, "")
+		finished, err := isJobFinished(job)
 		if err != nil {
-			err = errors.Wrap(err, "set status")
+			statusState = api.RestoreFailed
+			statusMsg = err.Error()
 			return rr, err
 		}
-	}
+		if !finished {
+			log.Info("Waiting for restore job to finish", "job", job.Name)
+			return rr, nil
+		}
 
-	err = r.startCluster(clusterOrig)
-	if err != nil {
-		err = errors.Wrap(err, "restart cluster")
-		return rr, err
-	}
+		if cr.Spec.PITR != nil {
+			if cluster.Spec.Pause {
+				err = k8sretry.RetryOnConflict(k8sretry.DefaultRetry, func() error {
+					current := new(api.PerconaXtraDBCluster)
+					err := r.client.Get(ctx, types.NamespacedName{Name: cluster.Name, Namespace: cluster.Namespace}, current)
+					if err != nil {
+						return errors.Wrap(err, "get cluster")
+					}
+					current.Spec.Pause = false
+					current.Spec.PXC.Size = 1
+					current.Spec.AllowUnsafeConfig = true
+					return r.client.Update(ctx, current)
+				})
+				if err != nil {
+					return rr, errors.Wrap(err, "update cluster")
+				}
+				return rr, nil
+			} else {
+				if cluster.Status.ObservedGeneration == cluster.Generation && cluster.Status.PXC.Status == api.AppStateReady {
```

Comment:
not sure about this condition. why do we say `waiting for cluster to start` only if `cluster.Status.PXC.Status` is ready?

Reply:
https://github.com/percona/percona-xtradb-cluster-operator/pull/1610/commits/7b5a420742ce91caf7574135e02a02c633104621